src/util-profiling.c

```diff
@@ -1245,7 +1245,497 @@ int SCProfileRuleStart(Packet *p)
+    if (tp > SC_ATOMIC_GET(proflite_tp_entry)) {
+        SC_ATOMIC_SET(proflite_tp_exit, tp);
+        SCLogNotice("exit point set to \"%s\" Tp now %u", setting, SC_ATOMIC_GET(proflite_tp_exit));
+    } else {
+        SCLogNotice("tracepoint should be > entry (TODO error handle)");
+    }
+}
+
+void ProfliteEnable(const char *setting)
+{
+    if (strcmp(setting, "all") == 0) {
+        SC_ATOMIC_OR(proflite_features, PROFLITE_ALL_BIT);
+    } else if (strcmp(setting, "tcp") == 0) {
+        SC_ATOMIC_OR(proflite_features, PROFLITE_TCP_BIT);
+    } else if (strcmp(setting, "app_http") == 0) {
+        SC_ATOMIC_OR(proflite_features, PROFLITE_ALPROTO_HTTP_BIT);
+    } else if (strcmp(setting, "only_app_http") == 0) {
+        SC_ATOMIC_SET(proflite_features, PROFLITE_ALPROTO_HTTP_BIT);
+    } else if (strcmp(setting, "app_dns") == 0) {
+        SC_ATOMIC_OR(proflite_features, PROFLITE_ALPROTO_DNS_BIT);
+    } else if (strcmp(setting, "only_app_dns") == 0) {
+        SC_ATOMIC_SET(proflite_features, PROFLITE_ALPROTO_DNS_BIT);
+    } else if (strcmp(setting, "app_ftp") == 0) {
+        SC_ATOMIC_OR(proflite_features, PROFLITE_ALPROTO_FTP_BIT);
+    } else if (strcmp(setting, "only_app_ftp") == 0) {
+        SC_ATOMIC_SET(proflite_features, PROFLITE_ALPROTO_FTP_BIT);
+    } else if (strcmp(setting, "app_dcerpc") == 0) {
+        SC_ATOMIC_OR(proflite_features, PROFLITE_ALPROTO_DCERPC_BIT);
+    } else if (strcmp(setting, "only_app_dcerpc") == 0) {
+        SC_ATOMIC_SET(proflite_features, PROFLITE_ALPROTO_DCERPC_BIT);
+    } else if (strcmp(setting, "any") == 0) {
+        SC_ATOMIC_SET(proflite_features, UINT64_MAX);
+    } else {
+        SCLogNotice("unknown setting %s", setting);
+        return;
+    }
+    SCLogNotice("enabled \"%s\". Flags now %" PRIx64, setting, SC_ATOMIC_GET(proflite_features));
+}
+
+void ProfliteDisable(const char *setting)
+{
+#if 0
+    if (strcmp(setting, "global") == 0) {
+        SC_ATOMIC_AND(proflite_flags, ~PROFLITE_ENABLED_BIT);
+    } else if (strcmp(setting, "app_http") == 0) {
+        SC_ATOMIC_AND(proflite_flags, ~PROFLITE_ALPROTO_HTTP_BIT);
+    } else {
+        abort();
+    }
+    SCLogNotice("disabled \"%s\". Flags now %"PRIx64, setting, SC_ATOMIC_GET(proflite_flags));
+#endif
+}
+
+enum ProfileLiteTracker {
+    PLT_ALL,
+    PLT_ALL_PSEUDO,
+    PLT_ALL_PAYLOAD,
+    PLT_ALL_NOPAYLOAD,
+    PLT_ALL_ALERT,
+    PLT_ALL_NOALERT,
+    PLT_TCP_ALL,
+    PLT_TCP_SYN,
+    PLT_TCP_RST,
+    PLT_TCP_FIN,
+    PLT_TCP_OTHER,
+    PLT_UDP,
+    PLT_ICMP4,
+    PLT_ICMP6,
+    PLT_OTHERIP,
+    PLT_OTHER,
+    PLT_ALPROTO_HTTP,
+    PLT_ALPROTO_HTTP2,
+    PLT_ALPROTO_SMB,
+    PLT_ALPROTO_NFS,
+    PLT_ALPROTO_DNS,
+    PLT_ALPROTO_DCERPC,
+    PLT_ALPROTO_SMTP,
+    PLT_ALPROTO_FTP,
+    PLT_ALPROTO_FTPDATA,
+    PLT_ALPROTO_TLS,
+    PLT_ALPROTO_SSH,
+    PLT_ALPROTO_DHCP,
+    PLT_ALPROTO_SNMP,
+    PLT_ALPROTO_TFTP,
+    PLT_ALPROTO_OTHER,
+    PLT_ALPROTO_NONE,
+    PLT_SIZE,
+};
```

Comment:
Any thought how dynamically loaded protocol parsers could be handled? Just as PLT_ALPROTO_OTHER?